templates/grub.cfg.erb

```diff
@@ -0,0 +1,8 @@
+insmod regexp
+
+set net_pxe_mac=$net_default_mac
+regexp --set=1:m1 --set=2:m2 --set=3:m3 --set=4:m4 --set=5:m5 --set=6:m6 '^([0-9a-f]{1,2})\:([0-9a-f]{1,2})\:([0-9a-f]{1,2})\:([0-9a-f]{1,2})\:([0-9a-f]{1,2})\:([0-9a-f]{1,2})' "$net_default_mac"
+mac_dashed=${m1}-${m2}-${m3}-${m4}-${m5}-${m6}
+
+mac_configfile=$prefix/grub.cfg-01-$mac_dashed
+source "$mac_configfile"
```

Comment:
Grub2 Ubuntu 16.04 lookup into "grub" directory whereas foreman create the associate host file in "grub2".
Quick workaround for both issues is to use Fedora grub2x64.efi instead, thanks to lzap!